src/ros_filter.cpp

```diff
@@ -1400,18 +1407,48 @@ void RosFilter<T>::loadParams()
       // update configuration (as this contains pose information)
       std::vector<bool> update_vec = loadUpdateConfig(imu_topic_name);
 
+      // sanity checks for update config settings
+      std::vector<int> position_update_vec(update_vec.begin() + POSITION_OFFSET,
+        update_vec.begin() + POSITION_OFFSET + POSITION_SIZE);
+      int position_update_sum = std::accumulate(position_update_vec.begin(),
+          position_update_vec.end(), 0);
+      if (position_update_sum > 0) {
+        RCLCPP_WARN(this->get_logger(),
+          "Warning: Some position entries in parameter %s_config are listed "
+          "true, but sensor_msgs/Imu contains no information about position",
+          imu_topic_name);
+      }
+      std::vector<int> linear_velocity_update_vec(
+        update_vec.begin() + POSITION_V_OFFSET,
+        update_vec.begin() + POSITION_V_OFFSET + LINEAR_VELOCITY_SIZE);
+      int linear_velocity_update_sum = std::accumulate(
+        linear_velocity_update_vec.begin(), linear_velocity_update_vec.end(),
+        0);
+      if (linear_velocity_update_sum > 0) {
+        RCLCPP_WARN(this->get_logger(),
+          "Warning: Some linear velocity entries in parameter %s_config are "
+          "listed true, but an sensor_msgs/Imu contains no information about "
+          "linear velocities", imu_topic_name);
+      }
+
       std::vector<bool> pose_update_vec = update_vec;
+      // IMU message contains no information about position, filter everything
+      // except orientation
+      std::fill(pose_update_vec.begin() + POSITION_OFFSET,
+        pose_update_vec.begin() + POSITION_OFFSET + POSITION_SIZE, 0);
       std::fill(pose_update_vec.begin() + POSITION_V_OFFSET,
         pose_update_vec.begin() + POSITION_V_OFFSET + TWIST_SIZE, 0);
       std::fill(pose_update_vec.begin() + POSITION_A_OFFSET,
-        pose_update_vec.begin() + POSITION_A_OFFSET + ACCELERATION_SIZE,
-        0);
+        pose_update_vec.begin() + POSITION_A_OFFSET + ACCELERATION_SIZE, 0);
 
       std::vector<bool> twist_update_vec = update_vec;
+      // IMU message contains no information about linear speeds, filter
+      // everything except angular velocity
       std::fill(twist_update_vec.begin() + POSITION_OFFSET,
         twist_update_vec.begin() + POSITION_OFFSET + POSE_SIZE, 0);
-      std::fill(
-        twist_update_vec.begin() + POSITION_A_OFFSET,
+      std::fill(twist_update_vec.begin() + POSITION_V_OFFSET,
```

Comment:
Do you know why we're changing the fill here? This has a pretty different effect, just making sure that its what we should be doing.

Reply:
I don't really know what's going on with most of these ports either. This one comes from this commit https://github.com/cra-ros-pkg/robot_localization/commit/0d32daf42d4b019a2d180ac53e01582cf22165b7

Reply:
This was a fix for an issue whereby we were fusing non-existent data in IMU messages:

https://github.com/cra-ros-pkg/robot_localization/issues/432